application/src/main/java/org/thingsboard/server/service/transport/DefaultTransportApiService.java

```diff
@@ -304,6 +304,17 @@ private BasicCredentialsValidationResult validateMqttCredentials(TransportProtos
                     TbMsg tbMsg = TbMsg.newMsg(DataConstants.ENTITY_CREATED, deviceId, customerId, metaData, TbMsgDataType.JSON, mapper.writeValueAsString(entityNode));
                     tbClusterService.pushMsgToRuleEngine(tenantId, deviceId, tbMsg, null);
                 }
+
+                List<EntityRelation> currentLastConnectedGatewayRelationList = relationService.findByFromAndType(TenantId.SYS_TENANT_ID, device.getId(), DataConstants.LAST_CONNECTED_GATEWAY, RelationTypeGroup.COMMON);
```

Comment:
Are you 100% sure this logic covers common cases? Maybe customers want to send data to all gateways? 
I believe a much easier option is to create a relation for each gateway. Maybe we do need a configuration parameter that forces the creation of these relations. Because by default, single relation "Created" is sufficient.